transport/server.cc

```diff
@@ -713,8 +713,10 @@ future<> cql_server::connection::process_request() {
         auto mem_estimate = f.length * 2 + 8000; // Allow for extra copies and bookkeeping
 
         if (mem_estimate > _server._max_request_size) {
-            return make_exception_future<>(exceptions::invalid_request_exception(format("request size too large (frame size {:d}; estimate {:d}; allowed {:d}",
-                    f.length, mem_estimate, _server._max_request_size)));
+            return _read_buf.skip(f.length).then([length = f.length, mem_estimate, this] {
```

Comment:
Looks good. The need to duplicate this skip() part twice made me think if perhaps the caller, upon getting any exception (including these two, but perhaps others), should be the one who skips the full request. But it will require knowing where the request will end (we may even get an exception in the middle of reading a request), so it might not be easy. So we can start with this code and rethink this later if we want.